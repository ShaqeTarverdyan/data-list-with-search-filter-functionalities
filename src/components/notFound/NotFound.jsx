import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Wrapper = styled.div`
  width: 100%;
  height: 12rem;
  display: flex;
  align-items: center;
  justify-content: center;
  color: var(--color-teal);
  background: var(--color-gray);
  box-shadow: 0 2px 3px 0 #e8ebed, 0 0 3px 0 #e8ebed;
`;

const NotFound = ({ text }) => {
  return <Wrapper data-testid="not-found">{text}</Wrapper>;
};

export default NotFound;

NotFound.propTypes = {
  text: PropTypes.string,
};
