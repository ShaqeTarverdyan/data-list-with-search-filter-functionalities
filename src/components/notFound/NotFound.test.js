import React from "react";
import NotFound from "./NotFound";

import { render } from "@testing-library/react";

const mockedText = "There is no any matched data";

describe("NotFound", () => {
  test("renders NotFound component", () => {
    render(<NotFound text={mockedText} />);
  });
});
