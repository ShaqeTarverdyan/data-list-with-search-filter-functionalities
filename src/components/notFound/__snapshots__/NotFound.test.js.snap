// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL bPOXGw"
  >
    There is no any matched data
  </div>
</DocumentFragment>
`;

exports[`testing the render of the NotFound component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL bPOXGw"
  >
    There is no any matched data
  </div>
</DocumentFragment>
`;
